### DISCUSSION TITLE
Partial updates of entities with collections

### DISCUSSION THREAD

**Discussion by MichaelM223** [leonardoporro/Detached-Mapper#82] [Ideas]
I was struggling with updating some entities with collections that have their collection set to `null` when relationships have not been resolved yet. EF Core skips `null` collections for when checking for relationship modifications.
`EntityCollectionTypeMapper` and `MergeEntityCollectionTypeMapper' however map `null` values to an empty collection.

With non-collection members I can agree you can't differ from removal or not set when a `null` value is passed. With collections though I think a empty collection indicates deletion of all relationships and `null` is clearly 'not-set' in my opinion.

The changes below work perfectly in my case but maybe I'm forgetting something?

Would it be ok if I submitted a pull request for two small changes in `EntityCollectionTypeMapper` and `MergeEntityCollectionTypeMapper'?
After line 31 I would add:
```csharp
if (source == null)
{
    return target;
}
```
https://github.com/leonardoporro/Detached-Mapper/blob/66bd47538c4b0dbfc0f6bba904e107c9e300be96/src/Detached.Mappers/TypeMappers/Entity/Collection/EntityCollectionTypeMapper.cs#L30-L33

https://github.com/leonardoporro/Detached-Mapper/blob/66bd47538c4b0dbfc0f6bba904e107c9e300be96/src/Detached.Mappers/TypeMappers/Entity/Collection/MergeEntityCollectionTypeMapper.cs#L30-L33

**Comment by leonardoporro** [comment]
Makes sense, but I would do it configurable just in case it breaks scenarios for other few people who is using it

**Reply by MichaelM223** [reply]
Good idea, I'll try to figure out something configurable and make a pull request

**Reply by MichaelM223** [reply]
Took a while but finally had time create a pull request (#100)

**Reply by leonardoporro** [reply]
Thank you! 
Sorry, I will have time to check on it this weekend. i.e.: tomorrow night or on Saturday

**Comment by leonardoporro** [comment]
I manually added the feature and it's called MapParameters.CompositeCollectionBehavior
And the values are Append and Merge.. where Append ignore nulls and Merge will delete the DB.
Does this make sense?

**Reply by leonardoporro** [reply]
It's 4am here.. so maybe it doesn't
